
# HG changeset patch
# User Hans Verkuil <hverkuil@xs4all.nl>
# Date 1258283654 -3600
# Node ID f995dea9e4698c69c98ee036b17942a82d5bf0bd
# Parent  b2210679269acd93375d05f38b89b5a3728f19d5
s2250: Mutex function usage.

From: Pete Eberlein <pete@sensoray.com>

Fix mutex function usage, which was overlooked in a previous patch.

Priority: high

Signed-off-by: Pete Eberlein <pete@sensoray.com>
Signed-off-by: Hans Verkuil <hverkuil@xs4all.nl>

--- a/drivers/staging/go7007/s2250-board.c
+++ b/drivers/staging/go7007/s2250-board.c
@@ -261,7 +261,7 @@ static int read_reg_fp(struct i2c_client
 
 	memset(buf, 0xcd, 6);
 	usb = go->hpi_context;
-	if (down_interruptible(&usb->i2c_lock) != 0) {
+	if (mutex_lock_interruptible(&usb->i2c_lock) != 0) {
 		printk(KERN_INFO "i2c lock failed\n");
 		kfree(buf);
 		return -EINTR;
@@ -270,7 +270,7 @@ static int read_reg_fp(struct i2c_client
 		kfree(buf);
 		return -EFAULT;
 	}
-	up(&usb->i2c_lock);
+	mutex_unlock(&usb->i2c_lock);
 
 	*val = (buf[0] << 8) | buf[1];
 	kfree(buf);
